### PR TITLE
Feature/sass variable error fix

### PR DESCRIPTION
**Pull request by ashblue**
Here is the fix for the recursive compiling, styling crashes, and assembler to styles coupling. Everything is now dumped in a cache folder for post processing the SCSS and JSON.

PLEASE NOTE: Make sure to run this locally and verify everything compiles okay. I put some extra QA effort into this. But please verify just to be sure. I'm always nervous about big build pipeline changes across different computers.

### CODE REVIEW

File: gulp/config.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  },
+  templates: {
+    watch: "src/**/*.{html,md,json,yml}"
+  },
+  dest: "dist",
+  cache: 'cache',
+  variables: {
+    jsonVariables: "cache/variables.json",
+    sassVariables:
+      "src/assets/toolkit/styles/partials/build/_import-variables.scss",
```
Comment (zsteiner) [inline gulp/config.js]: We should update these or remove?

File: gulp/config.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    partials: "src/assets/toolkit/styles/partials/build",
```
Comment (zsteiner) [inline gulp/config.js]: Ditto above comment.

File: gulp/config.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    dataBuild: "src/data/build/"
```
Comment (zsteiner) [inline gulp/config.js]: Update or remove

File: gulp/config.js (hunk 1 above)
Comment (ashblue) [reply gulp/config.js]: Good point, I'll strip them out.

**Pull request merged**